tests/mpi/test_checkpoints.py

```diff
@@ -0,0 +1,234 @@
+import copy
+import dataclasses
+import os
+from typing import List, Tuple
+
+import dacite
+import f90nml
+import xarray as xr
+import yaml
+
+import fv3core
+import pace.dsl
+import pace.util
+from fv3core.initialization.dycore_state import DycoreState
+from fv3core.testing.translate_fvdynamics import TranslateFVDynamics
+from pace.stencils.testing import TranslateGrid, dataset_to_dict
+from pace.stencils.testing.grid import Grid
+from pace.util.grid import DampingCoefficients, GridData
+from pace.util.testing import perturb
+
+
+def get_grid(data_path: str, rank: int, layout: Tuple[int, int], backend: str) -> Grid:
+    ds_grid: xr.Dataset = xr.open_dataset(os.path.join(data_path, "Grid-Info.nc")).isel(
+        savepoint=0
+    )
+    grid = TranslateGrid(
+        dataset_to_dict(ds_grid.isel(rank=rank)),
+        rank=rank,
+        layout=layout,
+        backend=backend,
+    ).python_grid()
+    return grid
+
+
+class StateInitializer:
+    def __init__(
+        self,
+        ds: xr.Dataset,
+        translate: TranslateFVDynamics,
+        communicator,
+        stencil_factory,
+        grid,
+        dycore_config,
+        consv_te,
+        n_split,
+    ):
+        self._ds = ds
+        self._translate = translate
+        self._consv_te = consv_te
+        self._n_split = n_split
+
+        input_data = dataset_to_dict(self._ds.copy())
+        state, grid_data = self._translate.prepare_data(input_data)
+        self._dycore = fv3core.DynamicalCore(
+            comm=communicator,
+            grid_data=grid_data,
+            stencil_factory=stencil_factory,
+            damping_coefficients=grid.damping_coefficients,
+            config=dycore_config,
+            phis=state.phis,
+            state=state,
+        )
+
+    def new_state(self) -> Tuple[DycoreState, GridData]:
+        input_data = dataset_to_dict(self._ds.copy())
+        state, grid_data = self._translate.prepare_data(input_data)
+
+        self._dycore.update_state(
+            self._consv_te,
+            input_data["do_adiabatic_init"],
+            input_data["bdt"],
+            self._n_split,
+            state,
+        )
+        return state, grid_data
+
+
+def test_fv_dynamics(
+    backend: str, data_path: str, calibrate_thresholds: bool, threshold_path: str
+):
+    print("start test call")
+    namelist = pace.util.Namelist.from_f90nml(
+        f90nml.read(os.path.join(data_path, "input.nml"))
+    )
+    threshold_filename = os.path.join(threshold_path, "fv_dynamics.yaml")
+    communicator = pace.util.CubedSphereCommunicator(
```

Comment:
Ideally we should refactor this initialization logic into one or more functions that gets re-used by the driver.